starlite/routes/asgi.py

```diff
@@ -27,12 +27,13 @@ def __init__(
             path: The path for the route.
             route_handler: An instance of [ASGIRouteHandler][starlite.handlers.asgi.ASGIRouteHandler].
         """
-        self.route_handler = route_handler
         super().__init__(
             path=path,
             scope_type=ScopeType.ASGI,
             handler_names=[unwrap_partial(route_handler.handler_name)],
         )
+        self.route_handler = route_handler
+        self.is_mount = route_handler.is_mount
```

Comment:
The alternative to defining `is_mount` here and on `BaseRoute` was this line in `add_route_to_trie`:

```python
is_mount = hasattr(route, "route_handler") and getattr(route.route_handler, "is_mount", False) # pyright: ignore
```
Just thought it cleaned it up a bit.